spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/jpa/JpaRepositoriesAutoConfiguration.java

```diff
@@ -66,10 +69,23 @@
 @ConditionalOnMissingBean({ JpaRepositoryFactoryBean.class, JpaRepositoryConfigExtension.class })
 @ConditionalOnProperty(prefix = "spring.data.jpa.repositories", name = "enabled", havingValue = "true",
 		matchIfMissing = true)
-@Import(JpaRepositoriesRegistrar.class)
 @AutoConfigureAfter({ HibernateJpaAutoConfiguration.class, TaskExecutionAutoConfiguration.class })
 public class JpaRepositoriesAutoConfiguration {
 
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnClass(EnversRevisionRepositoryFactoryBean.class)
```

Comment:
This conditional causes all the Spring Boot tests to use the Spring Data Envers auto-configuration implicitly as the dependency is always in the classpath (except when the `FilteredClassLoader` is used).

Is it desired or can it be an issue? Any suggestion to have something better?